### ISSUE TITLE
[Bug]: /pregen command fails to disable Server Watchdog, crashes server

### ISSUE THREAD

**Issue by clayplumridge** [FTBTeam/FTB-Modpack-Issues#1102]
### Modpack

FTB Plexiglass Mountain - 1.1.1

### Modpack version

1.1.1

### Log Files

https://pste.ch/senapaxudu.sql

### Describe the bug

When attempting to use the `/pregen` command, the server crashes after about a a minute. From reading the logs, it looks like MineTogether is failing to kill / suspend the Server Watchdog; see relevant logs below.

### Steps to reproduce

1. Login to the server on an op account
2. Run `/pregen overworld 0 5000 0 5000 1 true`
3. Observe a timeout on the client
4. Observe the server crashes

### Expected behaviour

Not crashing?

### Additional information

Most relevant section of the logs, from near the end:

```
[11Jul2022 14:36:33.859] [Server thread/INFO] [net.creeperhost.minetogether.MineTogetherCommon/]: We're about to kill the Server Watchdog. Don't worry, we'll resuscitate it! The next error is normal.
[11Jul2022 14:37:32.644] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[11Jul2022 14:37:32.644] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[11Jul2022 14:37:32.644] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[11Jul2022 14:37:32.648] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: Considering it to be crashed, server will forcibly shutdown.
[11Jul2022 14:37:32.644] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[11Jul2022 14:37:32.653] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: Considering it to be crashed, server will forcibly shutdown.
[11Jul2022 14:37:32.644] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[11Jul2022 14:37:32.653] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: Considering it to be crashed, server will forcibly shutdown.
[11Jul2022 14:37:32.646] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[11Jul2022 14:37:32.646] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: Considering it to be crashed, server will forcibly shutdown.
[11Jul2022 14:37:32.646] [Server Watchdog/ERROR] [net.minecraft.server.dedicated.ServerWatchdog/FATAL]: Considering it to be crashed, server will forcibly shutdown.
```

**Comment by clayplumridge** [comment]
@Saereth Here ya go! https://www.toptal.com/developers/hastebin/ahoyipamid.pl

**Comment by Saereth** [comment]
Mmm yeah fun. this one is more than a pack issue, gonna have to get this over to the Minetogether devs, thanks for reporting!

**Comment by TheonlyTazz** [comment]
In the meantime, u can set max-tick-time to -1 in your server.properties and pregen, this is basically the manual way to disable watchdog

**Comment by Raidobw2** [comment]
I'm unsure if the MT devs will be able to "fix" that, since it looks to involve some code "hacks" :o (killing threads!). Setting the watchdog to -1 like Tazz mentioned should normally be the intended solution for any server op. I'll report it to MT in any case, maybe they'll be able to look into it.

Additionally you can use an alternative:

> Try `/forge generate`
> E.g. `/forge generate ~ ~ ~ 62500 minecraft:overworld` looks like it will generate around a radius of 2000 blocks from your current player position.
> If you want to generate a radius of 3600 blocks instead:
> 3600 / 16 = a radius of 225 chunks
> 225 * 2 = 450 chunks in diameter
> The area of that square to generate is the diameter * diameter:
> 450 * 450 = 202'500 chunks in total
> `/forge generate ~ ~ ~ 202500 minecraft:overworld`
> Complex and undocumented but I think that should be right

TLDR
Pregen a radius of 2000 blocks: `/forge generate ~ ~ ~ 62500 minecraft:overworld`
Pregen a radius of 3600 blocks: `/forge generate ~ ~ ~ 202500 minecraft:overworld`
Pregen a radius of 3600+ block: See logic above and you will need to do a tiny bit of math :)

I hope that helps
